helixbass/tsc-rust
# Parse, bind + type-check var/ambient declarations

In this PR:
- parse, bind + type-check `var` declarations
- parse, bind + type-check "ambient" (`declare`) declarations

To test:
If you run `cargo run tmp.tsx` against a source file containing eg `var x: number = true;` then you should see a diagnostic debug-logged with message `Type 'boolean' is not assignable to type 'number'.`
If you run `cargo run tmp.tsx` against a source file containing eg:
```
declare var NaN: number;

const x: boolean = NaN;
```
then you should see a diagnostic debug-logged with message `Type 'number' is not assignable to type 'boolean'.`. In the `post-binding:` debug-logged `SourceFile` on the `VariableStatement`'s `BaseNode` you should see `modifiers` populated with a `DeclareKeyword` (and `parent` binding should be initialized)

Based on `parse-line-comment`
